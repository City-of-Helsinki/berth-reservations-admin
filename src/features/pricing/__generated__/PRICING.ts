/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { PriceUnits, ProductServiceType, PeriodType, AdditionalProductType, AdditionalProductTaxEnum } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: PRICING
// ====================================================

export interface PRICING_berthPriceGroups_edges_node_defaultProduct {
  __typename: "BerthProductNode";
  priceUnit: PriceUnits;
  priceValue: any;
}

export interface PRICING_berthPriceGroups_edges_node {
  __typename: "BerthPriceGroupNode";
  id: string;
  name: string;
  defaultProduct: PRICING_berthPriceGroups_edges_node_defaultProduct | null;
}

export interface PRICING_berthPriceGroups_edges {
  __typename: "BerthPriceGroupNodeEdge";
  node: PRICING_berthPriceGroups_edges_node | null;
}

export interface PRICING_berthPriceGroups {
  __typename: "BerthPriceGroupNodeConnection";
  edges: (PRICING_berthPriceGroups_edges | null)[];
}

export interface PRICING_winterStorageAreas_edges_node_properties_product {
  __typename: "WinterStorageProductNode";
  priceValue: any;
  priceUnit: PriceUnits;
}

export interface PRICING_winterStorageAreas_edges_node_properties {
  __typename: "WinterStorageAreaProperties";
  name: string | null;
  product: PRICING_winterStorageAreas_edges_node_properties_product | null;
}

export interface PRICING_winterStorageAreas_edges_node {
  __typename: "WinterStorageAreaNode";
  id: string;
  properties: PRICING_winterStorageAreas_edges_node_properties | null;
}

export interface PRICING_winterStorageAreas_edges {
  __typename: "WinterStorageAreaNodeEdge";
  node: PRICING_winterStorageAreas_edges_node | null;
}

export interface PRICING_winterStorageAreas {
  __typename: "WinterStorageAreaNodeConnection";
  edges: (PRICING_winterStorageAreas_edges | null)[];
}

export interface PRICING_additionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface PRICING_additionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: PRICING_additionalProducts_edges_node | null;
}

export interface PRICING_additionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (PRICING_additionalProducts_edges | null)[];
}

export interface PRICING_optionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface PRICING_optionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: PRICING_optionalProducts_edges_node | null;
}

export interface PRICING_optionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (PRICING_optionalProducts_edges | null)[];
}

export interface PRICING {
  berthPriceGroups: PRICING_berthPriceGroups | null;
  winterStorageAreas: PRICING_winterStorageAreas | null;
  additionalProducts: PRICING_additionalProducts | null;
  optionalProducts: PRICING_optionalProducts | null;
}
