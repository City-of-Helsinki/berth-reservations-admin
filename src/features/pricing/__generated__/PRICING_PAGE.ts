/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { PriceUnits, ProductServiceType, PeriodType, AdditionalProductType, AdditionalProductTaxEnum } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: PRICING_PAGE
// ====================================================

export interface PRICING_PAGE_berthPriceGroups_edges_node_defaultProduct {
  __typename: "BerthProductNode";
  priceUnit: PriceUnits;
  priceValue: any;
}

export interface PRICING_PAGE_berthPriceGroups_edges_node {
  __typename: "BerthPriceGroupNode";
  id: string;
  name: string;
  defaultProduct: PRICING_PAGE_berthPriceGroups_edges_node_defaultProduct | null;
}

export interface PRICING_PAGE_berthPriceGroups_edges {
  __typename: "BerthPriceGroupNodeEdge";
  node: PRICING_PAGE_berthPriceGroups_edges_node | null;
}

export interface PRICING_PAGE_berthPriceGroups {
  __typename: "BerthPriceGroupNodeConnection";
  edges: (PRICING_PAGE_berthPriceGroups_edges | null)[];
}

export interface PRICING_PAGE_winterStorageAreas_edges_node_properties_product {
  __typename: "WinterStorageProductNode";
  priceValue: any;
  priceUnit: PriceUnits;
}

export interface PRICING_PAGE_winterStorageAreas_edges_node_properties {
  __typename: "WinterStorageAreaProperties";
  name: string | null;
  product: PRICING_PAGE_winterStorageAreas_edges_node_properties_product | null;
}

export interface PRICING_PAGE_winterStorageAreas_edges_node {
  __typename: "WinterStorageAreaNode";
  id: string;
  properties: PRICING_PAGE_winterStorageAreas_edges_node_properties | null;
}

export interface PRICING_PAGE_winterStorageAreas_edges {
  __typename: "WinterStorageAreaNodeEdge";
  node: PRICING_PAGE_winterStorageAreas_edges_node | null;
}

export interface PRICING_PAGE_winterStorageAreas {
  __typename: "WinterStorageAreaNodeConnection";
  edges: (PRICING_PAGE_winterStorageAreas_edges | null)[];
}

export interface PRICING_PAGE_additionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface PRICING_PAGE_additionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: PRICING_PAGE_additionalProducts_edges_node | null;
}

export interface PRICING_PAGE_additionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (PRICING_PAGE_additionalProducts_edges | null)[];
}

export interface PRICING_PAGE_optionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface PRICING_PAGE_optionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: PRICING_PAGE_optionalProducts_edges_node | null;
}

export interface PRICING_PAGE_optionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (PRICING_PAGE_optionalProducts_edges | null)[];
}

export interface PRICING_PAGE {
  berthPriceGroups: PRICING_PAGE_berthPriceGroups | null;
  winterStorageAreas: PRICING_PAGE_winterStorageAreas | null;
  additionalProducts: PRICING_PAGE_additionalProducts | null;
  optionalProducts: PRICING_PAGE_optionalProducts | null;
}
