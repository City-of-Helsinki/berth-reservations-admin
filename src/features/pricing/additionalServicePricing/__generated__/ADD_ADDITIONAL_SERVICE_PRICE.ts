/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CreateAdditionalProductMutationInput, PriceUnits, ProductServiceType, PeriodType, AdditionalProductTaxEnum, AdditionalProductType } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: ADD_ADDITIONAL_SERVICE_PRICE
// ====================================================

export interface ADD_ADDITIONAL_SERVICE_PRICE_createAdditionalProduct_additionalProduct {
  __typename: "AdditionalProductNode";
  id: string;
  priceValue: any;
  priceUnit: PriceUnits;
  service: ProductServiceType;
  period: PeriodType;
  taxPercentage: AdditionalProductTaxEnum;
  productType: AdditionalProductType;
}

export interface ADD_ADDITIONAL_SERVICE_PRICE_createAdditionalProduct {
  __typename: "CreateAdditionalProductMutationPayload";
  additionalProduct: ADD_ADDITIONAL_SERVICE_PRICE_createAdditionalProduct_additionalProduct | null;
}

export interface ADD_ADDITIONAL_SERVICE_PRICE {
  createAdditionalProduct: ADD_ADDITIONAL_SERVICE_PRICE_createAdditionalProduct | null;
}

export interface ADD_ADDITIONAL_SERVICE_PRICEVariables {
  input: CreateAdditionalProductMutationInput;
}
