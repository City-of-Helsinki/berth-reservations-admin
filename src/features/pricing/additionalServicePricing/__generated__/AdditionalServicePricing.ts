/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ProductServiceType, PriceUnits, PeriodType, AdditionalProductType, AdditionalProductTaxEnum } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: AdditionalServicePricing
// ====================================================

export interface AdditionalServicePricing_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
  taxPercentage: AdditionalProductTaxEnum;
}

export interface AdditionalServicePricing_edges {
  __typename: "AdditionalProductNodeEdge";
  node: AdditionalServicePricing_edges_node | null;
}

export interface AdditionalServicePricing {
  __typename: "AdditionalProductNodeConnection";
  edges: (AdditionalServicePricing_edges | null)[];
}
