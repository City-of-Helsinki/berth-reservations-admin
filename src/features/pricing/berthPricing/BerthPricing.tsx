import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Row } from 'react-table';
import { useMutation } from '@apollo/react-hooks';

import Card from '../../../common/card/Card';
import CardHeader from '../../../common/cardHeader/CardHeader';
import CardBody from '../../../common/cardBody/CardBody';
import Table, { Column, COLUMN_WIDTH } from '../../../common/table/Table';
import Section from '../../../common/section/Section';
import Text from '../../../common/text/Text';
import { formatPrice } from '../../../common/utils/format';
import EditForm, { EDIT_FORM_TYPE } from '../editModal/EditForm';
import { BerthPricing as BerthPricingData } from './__generated__/BerthPricing';
import { getBerthsData } from './utils';
import { PeriodType } from '../../../@types/__generated__/globalTypes';
import { getPeriodTKey } from '../../../common/utils/translations';
import EditModal from '../editModal/EditModal';
import { UPDATE_BERTH_PRICE_MUTATION, CREATE_BERTH_PRODUCT_MUTATION } from './mutations';
import {
  UPDATE_BERTH_PRICE,
  UPDATE_BERTH_PRICEVariables as UPDATE_BERTH_PRICE_VARS,
} from './__generated__/UPDATE_BERTH_PRICE';
import {
  CREATE_BERTH_PRODUCT,
  CREATE_BERTH_PRODUCTVariables as CREATE_BERTH_PRODUCT_VARS,
} from './__generated__/CREATE_BERTH_PRODUCT';

export interface BerthPrice {
  id: string;
  productId: string | undefined;
  name: string;
  privateCustomer: number | undefined;
  company: number | undefined;
  period: PeriodType;
}

export interface BerthPricingProps {
  className?: string;
  data: BerthPricingData | undefined | null;
  loading: boolean;
}

const BerthPricing = ({ className, data, loading }: BerthPricingProps) => {
  const { t, i18n } = useTranslation();
  const [editRowValues, setEditRowValues] = useState<BerthPrice>();
  const [updateBerthPrice] = useMutation<UPDATE_BERTH_PRICE, UPDATE_BERTH_PRICE_VARS>(UPDATE_BERTH_PRICE_MUTATION);
  const [createBerthPrice] = useMutation<CREATE_BERTH_PRODUCT, CREATE_BERTH_PRODUCT_VARS>(
    CREATE_BERTH_PRODUCT_MUTATION
  );

  const harborCols: Column<BerthPrice>[] = [
    {
      Header: t('pricing.berths.width') || '',
      accessor: 'name',
    },
    {
      Header: t('pricing.berths.privateCustomer') || '',
      accessor: 'privateCustomer',
      Cell: ({ cell: { value } }) => (value ? formatPrice(value, i18n.language) : '-'),
    },
    {
      Header: t('pricing.berths.company') || '',
      accessor: 'company',
      Cell: ({ cell: { value } }) => (value ? formatPrice(value, i18n.language) : '-'),
    },
    {
      Header: t('pricing.berths.period') || '',
      accessor: 'period',
      Cell: ({ cell }) => t(getPeriodTKey(cell.value)),
    },
    {
      id: 'edit',
      Header: t('common.edit') || '',
      sortType: 'none',
      width: COLUMN_WIDTH.S,
      Cell: ({ row }: { row: Row<BerthPrice> }) => (
        <button onClick={() => setEditRowValues(row.original)}>
          <Text color="brand">{t('common.edit')}</Text>
        </button>
      ),
    },
  ];

  const handleSubmit = async ({
    id: priceGroupId,
    productId: defaultProductId,
    privateCustomer: priceValue,
  }: BerthPrice): Promise<void> => {
    if (defaultProductId) {
      await updateBerthPrice({
        variables: { input: { id: defaultProductId, priceValue } },
      });
    } else {
      await createBerthPrice({
        variables: { input: { priceGroupId, priceValue } },
      });
    }

    setEditRowValues(undefined);
  };

  const handleClose = () => setEditRowValues(undefined);

  return (
    <>
      <Card className={className}>
        <CardHeader title={t('pricing.berths.title')} />
        <CardBody>
          <Section>{t('pricing.berths.description')}</Section>
          <Table
            columns={harborCols}
            data={getBerthsData(data)}
            loading={loading}
            theme="basic"
            renderEmptyStateRow={() => t('common.notification.noData.description')}
          />
        </CardBody>
      </Card>
      <EditModal isOpen={!!editRowValues} closeModal={handleClose}>
        {editRowValues && (
          <EditForm
            closeModal={handleClose}
            formType={EDIT_FORM_TYPE.BERTHS}
            initialValues={editRowValues}
            onSubmit={handleSubmit}
          />
        )}
      </EditModal>
    </>
  );
};

export default BerthPricing;
