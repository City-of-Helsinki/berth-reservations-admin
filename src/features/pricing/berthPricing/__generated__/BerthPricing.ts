/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { PriceUnits } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: BerthPricing
// ====================================================

export interface BerthPricing_edges_node_defaultProduct {
  __typename: "BerthProductNode";
  priceUnit: PriceUnits;
  priceValue: any;
}

export interface BerthPricing_edges_node {
  __typename: "BerthPriceGroupNode";
  id: string;
  name: string;
  defaultProduct: BerthPricing_edges_node_defaultProduct | null;
}

export interface BerthPricing_edges {
  __typename: "BerthPriceGroupNodeEdge";
  node: BerthPricing_edges_node | null;
}

export interface BerthPricing {
  __typename: "BerthPriceGroupNodeConnection";
  edges: (BerthPricing_edges | null)[];
}
