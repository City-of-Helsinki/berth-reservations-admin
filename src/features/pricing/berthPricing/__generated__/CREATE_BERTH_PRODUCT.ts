/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CreateBerthProductMutationInput, PriceUnits } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: CREATE_BERTH_PRODUCT
// ====================================================

export interface CREATE_BERTH_PRODUCT_createBerthProduct_berthProduct {
  __typename: "BerthProductNode";
  id: string;
  priceValue: any;
  priceUnit: PriceUnits;
}

export interface CREATE_BERTH_PRODUCT_createBerthProduct {
  __typename: "CreateBerthProductMutationPayload";
  berthProduct: CREATE_BERTH_PRODUCT_createBerthProduct_berthProduct | null;
}

export interface CREATE_BERTH_PRODUCT {
  createBerthProduct: CREATE_BERTH_PRODUCT_createBerthProduct | null;
}

export interface CREATE_BERTH_PRODUCTVariables {
  input: CreateBerthProductMutationInput;
}
