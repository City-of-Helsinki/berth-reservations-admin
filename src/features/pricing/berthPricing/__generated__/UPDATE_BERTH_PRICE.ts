/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UpdateBerthProductMutationInput, PriceUnits } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: UPDATE_BERTH_PRICE
// ====================================================

export interface UPDATE_BERTH_PRICE_updateBerthProduct_berthProduct {
  __typename: "BerthProductNode";
  id: string;
  priceValue: any;
  priceUnit: PriceUnits;
}

export interface UPDATE_BERTH_PRICE_updateBerthProduct {
  __typename: "UpdateBerthProductMutationPayload";
  berthProduct: UPDATE_BERTH_PRICE_updateBerthProduct_berthProduct | null;
}

export interface UPDATE_BERTH_PRICE {
  updateBerthProduct: UPDATE_BERTH_PRICE_updateBerthProduct | null;
}

export interface UPDATE_BERTH_PRICEVariables {
  input: UpdateBerthProductMutationInput;
}
