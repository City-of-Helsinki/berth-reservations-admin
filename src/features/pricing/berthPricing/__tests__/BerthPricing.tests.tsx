import React from 'react';
import { shallow } from 'enzyme';

import BerthPricing, { BerthPricingProps } from '../BerthPricing';
import { PriceUnits } from '../../../../@types/__generated__/globalTypes';

describe('BerthPricing', () => {
  const data: BerthPricingProps['data'] = {
    __typename: 'BerthPriceGroupNodeConnection',
    edges: [
      {
        __typename: 'BerthPriceGroupNodeEdge',
        node: {
          __typename: 'BerthPriceGroupNode',
          id: '68ac7db5-b397-48f6-af19-99c35d645cb6',
          name: 'sed sed debitis',
          defaultProduct: {
            __typename: 'BerthProductNode',
            priceValue: '10.2',
            priceUnit: PriceUnits.AMOUNT,
          },
        },
      },
    ],
  };

  const initialProps: BerthPricingProps = { data, loading: false, openModal: jest.fn() };

  const getWrapper = (props: Partial<BerthPricingProps> = {}) => shallow(<BerthPricing {...initialProps} {...props} />);

  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders noData message when there is no data', () => {
    const wrapper = getWrapper({ data: undefined });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
