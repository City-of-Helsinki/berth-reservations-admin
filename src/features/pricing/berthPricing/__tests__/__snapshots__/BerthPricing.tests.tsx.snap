// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthPricing renders noData message when there is no data 1`] = `"<div class=\\"card\\"><div class=\\"cardHeader\\"><span class=\\"title\\">Venepaikka</span></div><div class=\\"cardBody\\"><article class=\\"section\\"><section class=\\"body\\">Perushinta. Hinnat sisältävät alv 24 %.</section></article><div><div class=\\"tableWrapper\\"><div role=\\"table\\" style=\\"min-width:712.5px\\" class=\\"table basic noMainHeader\\"><div class=\\"stickyHeaders\\"><div role=\\"row\\" style=\\"display:flex;flex:1 0 auto;min-width:0px\\" class=\\"header\\"><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Leveys<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Yksityinen<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Yritys<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Aika<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px\\" class=\\"headerCell\\">Muokkaa</div></div></div><div role=\\"rowgroup\\"><div class=\\"rowWrapper\\" role=\\"row\\"><div class=\\"tableCell\\">Ei tietoja</div></div></div></div></div></div></div> </div>"`;

exports[`BerthPricing renders normally 1`] = `"<div class=\\"card\\"><div class=\\"cardHeader\\"><span class=\\"title\\">Venepaikka</span></div><div class=\\"cardBody\\"><article class=\\"section\\"><section class=\\"body\\">Perushinta. Hinnat sisältävät alv 24 %.</section></article><div><div class=\\"tableWrapper\\"><div role=\\"table\\" style=\\"min-width:712.5px\\" class=\\"table basic noMainHeader\\"><div class=\\"stickyHeaders\\"><div role=\\"row\\" style=\\"display:flex;flex:1 0 auto;min-width:0px\\" class=\\"header\\"><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Leveys<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Yksityinen<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Yritys<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px;cursor:pointer\\" title=\\"Toggle SortBy\\" class=\\"headerCell\\">Aika<div class=\\"sortArrowWrapper\\"><div class=\\"icon standard small sortArrow sortArrowDown\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"\\"><g><polygon points=\\"13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169\\" transform=\\"translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) \\"></polygon></g></svg></div></div></div><div colSpan=\\"1\\" role=\\"columnheader\\" style=\\"box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px\\" class=\\"headerCell\\">Muokkaa</div></div></div><div role=\\"rowgroup\\"><div class=\\"rowWrapper\\"><div role=\\"row\\" style=\\"display:flex;flex:1 0 auto;min-width:0px\\"><div role=\\"cell\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px\\" class=\\"tableCell\\">sed sed debitis</div><div role=\\"cell\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px\\" class=\\"tableCell\\">10,20 €</div><div role=\\"cell\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px\\" class=\\"tableCell\\">20,40 €</div><div role=\\"cell\\" style=\\"box-sizing:border-box;flex:150 0 auto;min-width:0px;width:150px\\" class=\\"tableCell\\">Kausi</div><div role=\\"cell\\" style=\\"box-sizing:border-box;flex:112.5 0 auto;min-width:0px;width:112.5px\\" class=\\"tableCell\\"><button><span class=\\"text brand span\\">Muokkaa</span></button></div></div></div></div></div></div></div></div> </div>"`;
